dev.py

```diff
@@ -1129,6 +1131,87 @@ def task_meta(cls, **kwargs):
         }
 
 
+@cli.cls_cmd('doctest')
+class Doctest(Task):
+    """
+    :wrench: Run doctests
+
+    \b
+    ```python
+    Examples:
+
+    $ python dev.py doctest -s {submodule}
+    $ python dev.py doctest -t {path/to/file}
+    ```
+    """
+    ctx = CONTEXT
+
+    doctests = Option(['--doctests'], default=True, is_flag=True)
+    submodule = Option(
+        ['--submodule', '-s'], default=None, metavar='MODULE_NAME',
+        help="Submodule whose tests to run (cluster, constants, ...)"
+    )
+    filename = Option(
+            ['--filename', '-t'], default=None, metavar='FILENAME',
+            help="Specify a .py file to check"
+        ) 
+    fail_fast = Option(
+        ['--fail-fast', '-x'], default=False, is_flag=True,
+        help="fail on first error"
+    )
+    verbose = Option(
+        ['--verbose', '-v'], default=False, is_flag=True,
+        help="verbosity"
+    )
+    
+    pytest_args = Argument(
+        ['pytest_args'], nargs=-1, metavar='PYTEST-ARGS', required=False
+    )
+
+    TASK_META = {
+        'task_dep': ['build'],
+    }
+
+    @classmethod
+    def scipy_doctests(cls, args, pytest_args):
+        dirs = Dirs(args)
+        dirs.add_sys_path()
+
+        extra_argv = pytest_args[:] if pytest_args else []
+        if extra_argv and extra_argv[0] == '--':
+            extra_argv = extra_argv[1:]
+
+        if args.fail_fast:
+            extra_argv += ['--fail-fast', args.fail_fast]
+
+        if args.submodule:
+            tests = [PROJECT_MODULE + "." + args.submodule]
+        elif args.filename:
+            tests = args.filename
+        else:
+            tests = None
+
+        runner, version, mod_path = get_test_runner(PROJECT_MODULE)
+
+        with working_dir(dirs.site):
+            print("Running doctests for {} version:{}, installed at:{}".format(
+                        PROJECT_MODULE, version, mod_path))
+            verbose = int(args.verbose) + 1
+            result = runner(  # scipy._lib._testutils:PytestTester
+                verbose=verbose,
+                extra_argv=extra_argv,
+                doctests=True,
+                tests=tests)
+        return result
+    
+    @classmethod
+    def run(cls, pytest_args, **kwargs):
+        kwargs.update(cls.ctx.get())
+        Args = namedtuple('Args', [k for k in kwargs.keys()])
+        args = Args(**kwargs)
+        return cls.scipy_doctests(args, pytest_args)
```

Comment:
I wonder if there's room for deduplicating `Test` and `Doctest` tasks.

Reply:
Let me look into it.

Reply:
Have you considered making this `python dev.py test --doctests` rather than a separate command? That way you may be able to just set the `pytest_args` you need, and not change anything else.

Reply:
This works. It does solve the duplication problem.